objc/objc_runtime_darwin.go

```diff
@@ -24,27 +24,41 @@ var (
 
 	objc_msgSend              = purego.Dlsym(objc, "objc_msgSend")
 	objc_msgSendSuper2        = purego.Dlsym(objc, "objc_msgSendSuper2")
```

Comment:
Don't we replace them with funcs?

Reply:
Their signatures are not known. They are determined by whichever ObjC method is being called. I didn't want to make any breaking changes to the exported api so I left them alone. How would we go about changing `ID.Send` and `ID.SendSuper` to utilize `purego.Func`?

Reply:
Sure. So can't we treat a variable number of function (`func(args ...interface{})`) with Func?

Reply:
Potentially, it would mean that `purego.Func` would need a special case for when the last argument is a slice of interfaces. I can draft something up tomorrow